### ISSUE TITLE
Add Django extension

### ISSUE THREAD

**Issue by rochacbruno** [rochacbruno/dynaconf#19]
A Django extension when enabled should replace `django.conf.settings` with Dynaconf instance

**Comment by rochacbruno** [comment]
There is a `dynaconf.contrib` with a Flask example

**Comment by rochacbruno** [comment]
Right now it is possible to use Dynaconf with django in a simple way by importing it on `settings.py`

In `settings.py` of django project one can do:

```py
from dynaconf import settings

VAR = settings.VAR
DEBUG = settings.as_bool('DEBUG')
ANOTHER_VALUE = settings.get('VALUE')
...
```

But it would be perfect if explicitly reading value by value is not needed and the `django.conf.settings` object can be `patched` and replaced by `dynaconf.settings`

This way it is possible to use fresh vars, casting, and all other dynaconf features.

The expected feature is by simply doing in a `app/views.py` in django

```py
from django.conf import settings   # this imports a `patched` `settings` object which is a `Dynaconf`

def index(request):
    settings.FOO   # this value is taken from dynaconf
    settings.get_fresh('BAR')   # this value is freshly taken from dynaconf (Redis etc...)
    ...
```

Right now my POC is to do the `patch` in `manage.py` or in `wsgi.py`


![dyna_django](https://user-images.githubusercontent.com/458654/31954848-6a1a14c8-b8c5-11e7-97f5-577aac1fd6a8.png)


Django has in `django/__init__.py` a `setup()` method which is the start point for manage and also wsgi, but this method is called only when it is started, I have not found a way to override it other than `hacking python module import hooks`

**Comment by rochacbruno** [comment]
Note that the idea is to **NOT TOUCH** Django's code, only **PROJECT** code, that is why `manage.py` and `wsgi.py` is the first approach.

**Comment by cuducos** [comment]
And old and ugly habit of Django developers is (was?) to add something like that at the bottom of the file to bring new and external stuff to it:


```python
try:
    from whatever_settings_module import *
except ImportError:
    pass
```

So we could offer something like that in `dynaconf.contrib` – a module to be imported that behaves as an injection in the namespace… have you tried something in that sense?

Ugly as hell… but just brainstorming here ; )

**Comment by renzon** [comment]
Well, once Django copy manger.py and wsgi.py to inside the project, they are project's files, not DJango's. if they hadn't want us hacking with those files, they would keep them as libs instead of copying them. Example, CLIng explicitly need to be inserted on https://github.com/renzon/wttd/blob/master/eventex/wsgi.py#L12. I'm not 100% sure once I am new to Django. my 2 cents

**Comment by rochacbruno** [comment]
@cuducos yeah that should work, in settings.py just `from dynaconf.settings import *` or something like this. But, the problem is that `dynaconf` settings are not `static`, the idea is that you can `export` new environment variables or change settings in a `redis` server and it is reflected in a running project `freshly`.

Normally we override `getitem` and `getattr` but it is not possible yet to do that for a python module as `settings.py` (the only way would be hacking python import hooks).


I will agree with @renzon and then use `wsgi.py` + `manage.py` as they are project files we can provide a `dj_dynaconf` extension to do that!

**Comment by bloodpet** [comment]
Won't `django.conf.settings.configure()` (see https://docs.djangoproject.com/en/1.11/topics/settings/#using-settings-without-setting-django-settings-module)  work for this?
Maybe calling `django.conf.settings.configure(**custom_settings)` inside an `apps.py` work. This way, there users won't need to add anything to `manage.py`, `wsgi.py`, and just add dynaconf django app to the `INSTALLED_APPS`.

**Comment by renzon** [comment]
Looked promising. But once settings is a dict (well, a **kwargs) it will lack the feature of getting dynamic config values mentioned on https://github.com/rochacbruno/dynaconf/issues/19#issuecomment-339081440.

**Comment by rochacbruno** [comment]
@renzon @bloodpet @cuducos 

I have it working now! going to write some tests and push!!

just add:

```py
INSTALLED_APPS = [
    'dynaconf.contrib.django_dynaconf',
    ..
]
```

and done!

Thanks you all for the ideas!

![dyna_django2](https://user-images.githubusercontent.com/458654/31965871-75b2e766-b8e7-11e7-88ea-7ba481fbff75.png)

**Comment by renzon** [comment]
I want to see the code, at least put it in a branch before running the tests =P

**Comment by rochacbruno** [comment]
@renzon https://github.com/rochacbruno/dynaconf/pull/34

**Comment by rochacbruno** [comment]
Implemented in #51
